src/browser/BrowserAction.cpp

```diff
@@ -465,6 +467,37 @@ QJsonObject BrowserAction::handleCreateNewGroup(const QJsonObject& json, const Q
     return buildResponse(action, message, newNonce);
 }
 
+QJsonObject BrowserAction::handleGetTotp(const QJsonObject& json, const QString& action)
+{
+    const QString nonce = json.value("nonce").toString();
+    const QString encrypted = json.value("message").toString();
+
+    if (!m_associated) {
+        return getErrorReply(action, ERROR_KEEPASS_ASSOCIATION_FAILED);
+    }
+
+    const QJsonObject decrypted = decryptMessage(encrypted, nonce);
+    if (decrypted.isEmpty()) {
+        return getErrorReply(action, ERROR_KEEPASS_CANNOT_DECRYPT_MESSAGE);
+    }
+
+    QString command = decrypted.value("action").toString();
+    if (command.isEmpty() || command.compare("get-totp", Qt::CaseSensitive) != 0) {
+        return getErrorReply(action, ERROR_KEEPASS_INCORRECT_ACTION);
+    }
+
+    const QString uuid = decrypted.value("uuid").toString();
+
+    // Get the current TOTP
+    const auto totp = browserService()->getCurrentTotp(uuid);
+    const QString newNonce = incrementNonce(nonce);
+
+    QJsonObject message = buildMessage(newNonce);
+    message["totp"] = totp;
```

Comment:
Since we return empty string if a totp is not found we should check for that on the extension side and warn the user that no TOTP was available. This might already be done.

Reply:
Good idea.